strax/corrections.py

```diff
@@ -38,31 +47,38 @@ def __init__(self, host='127.0 0.1', username=None, password=None,
         self.database = self.client[self.database_name]
 
     def list_corrections(self):
-        '''Smart logic to list all corrections.
-        '''
+        """Smart logic to list all corrections.
+        """
         return [x['name'] for x in self.database.list_collections()]
 
     def read(self, correction):
-        '''Smart logic to read corrections,
-        :param correction: correction's name (str type).
-        '''
+        """Smart logic to read corrections,
+        :param correction: pandas.DataFrame object name in the DB (str type).
+        """
         df = pdm.read_mongo(correction, [], self.database)
 
         # No data found
         if df.size == 0:
             return None
         # Delete internal Mongo identifier
         del df['_id']
+
+        df['time'] = pd.to_datetime(df['time'], utc=True)
+
         return df.set_index('time')
 
-    def interpolate(self, what, when, how='interpolate', **kwargs):
-        '''Interpolate values of a given correction
+    @staticmethod
+    def interpolate(what, when, how='interpolate', **kwargs):
+        """Interpolate values of a given correction
         For information of interpolation methods see,
         https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.interpolate.html
         :param what: what do you want to interpolate, what correction(DataFrame)
-        :param when: date,  e.g. datetime(2020, 8, 12, 21, 4, 32, 726110)
-        :param how: Interpolation method
-        '''
+        :param when: date, e.g. datetime(2020, 8, 12, 21, 4, 32, 7, tzinfo=pytz.utc)
+        :param how: Interpolation method, can be either how='interpolate' or how='fill'
+        :param kwargs: are forward to the interpolation
+        """
+        CorrectionsInterface.check_timezone(when)
+
         df = what
 
         df_new = pd.DataFrame.from_dict({'Time': [when]})
```

Comment:
Hej Aaron, looks great. Thanks for adding `check_timezone`. But beside just adding the function you should also explicitly check every `time`-input by the user. E.g. in this case here you should call `when = check_timezone(when)`. Otherwise the user can still put a wrong time without realizing it.

Reply:
The way he setup the check now it raises a ValueError if not in the correct timezone so this should work

Reply:
Totally agree, I have not seen that the function gets called at all.